packages/aws-cdk-lib/aws-codedeploy/test/server/deployment-group.test.ts

```diff
@@ -212,6 +213,197 @@ describe('CodeDeploy Server Deployment Group', () => {
+        'DeploymentStyle': {
+          'DeploymentOption': 'WITH_TRAFFIC_CONTROL',
+        },
+      });
+    } else if (TargetGroupInfoList.length > 0) {
+      Template.fromStack(stack).hasResourceProperties('AWS::CodeDeploy::DeploymentGroup', {
+        'LoadBalancerInfo': {
+          'TargetGroupInfoList': TargetGroupInfoList.map((tgName) => ({
+            'Name': {
+              'Fn::GetAtt': [
+                tgName,
+                'TargetGroupName',
+              ],
+            },
+          })),
+        },
+        'DeploymentStyle': {
+          'DeploymentOption': 'WITH_TRAFFIC_CONTROL',
+        },
+      });
+    } else if (ElbInfoList.length > 0) {
+      Template.fromStack(stack).hasResourceProperties('AWS::CodeDeploy::DeploymentGroup', {
+        'LoadBalancerInfo': {
+          'ElbInfoList': ElbInfoList.map((elbName) => ({
+            'Name': {
+              'Ref': elbName,
+            },
+          })),
+        },
+        'DeploymentStyle': {
+          'DeploymentOption': 'WITH_TRAFFIC_CONTROL',
+        },
+      });
+    }
+  });
+
+  test('can be created with multiple ALB Target Groups as the load balancers', () => {
+    const stack = new cdk.Stack();
+    const defaultVpc = new ec2.Vpc(stack, 'VPC');
+
+    const clb = new lbv1.LoadBalancer(stack, 'CLB', {
+      vpc: defaultVpc,
+    });
+
+    const alb = new lbv2.ApplicationLoadBalancer(stack, 'ALB', {
+      vpc: defaultVpc,
+    });
+
+    const nlb = new lbv2.NetworkLoadBalancer(stack, 'NLB', {
+      vpc: defaultVpc,
+    });
+
+    const listener = alb.addListener('Listener', { protocol: lbv2.ApplicationProtocol.HTTP });
+    const targetGroup = listener.addTargets('Fleet', { protocol: lbv2.ApplicationProtocol.HTTP });
+
+    const nlbListener = nlb.addListener('Listener', { port: 80 });
+    const nlbTargetGroup = nlbListener.addTargets('Fleet', { port: 80 });
+
+    new codedeploy.ServerDeploymentGroup(stack, 'DeploymentGroup', {
+      loadBalancers: [
+        codedeploy.LoadBalancer.classic(clb),
+        codedeploy.LoadBalancer.application(targetGroup),
+        codedeploy.LoadBalancer.network(nlbTargetGroup),
+      ],
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::CodeDeploy::DeploymentGroup', {
+      'LoadBalancerInfo': {
+        'TargetGroupInfoList': [
+          {
+            'Name': {
+              'Fn::GetAtt': [
+                'ALBListenerFleetGroup008CEEE4',
+                'TargetGroupName',
+              ],
+            },
+          },
+          {
+            'Name': {
+              'Fn::GetAtt': [
+                'NLBListenerFleetGroupB882EC86',
+                'TargetGroupName',
+              ],
+            },
+          },
+        ],
+        'ElbInfoList': [
+          {
+            'Name': {
+              'Ref': 'CLBA83A883E',
+            },
+          },
+        ],
+      },
+      'DeploymentStyle': {
+        'DeploymentOption': 'WITH_TRAFFIC_CONTROL',
+      },
+    });
+  });
```

Comment:
```suggestion
  test.each([
    ['CLB + ALB', true, true, false],
    ['CLB + NLB', true, false, true],
    ['ALB + NLB', false, true, true],
    ['CLB + ALB + NLB', true, true, true],
  ])('can be created with multiple load balancers (%s)', (_, withCLB, withALB, withNLB) => {
    const stack = new cdk.Stack();
    const defaultVpc = new ec2.Vpc(stack, 'VPC');

    const loadBalancers: codedeploy.LoadBalancer[] = [];
    const loadBalancerInfo: { TargetGroupInfoList?: any[]; ElbInfoList?: any[] } = {};

    if (withCLB) {
      loadBalancers.push(codedeploy.LoadBalancer.classic(new lbv1.LoadBalancer(stack, `CLB-${i}`, {
        vpc: defaultVpc,
        listeners: [{
          externalPort: 8080,
          externalProtocol: lbv1.LoadBalancingProtocol.TCP,
          internalProtocol: lbv1.LoadBalancingProtocol.TCP,
        }],
      })));

      loadBalancerInfo.ElbInfoList?.push({
        'Name': {
          'Ref': 'CLB677386E0',
        },
      });
    }

    if (withALB) {
      const alb = new lbv2.ApplicationLoadBalancer(stack, 'ALB', {
        vpc: defaultVpc,
      });

      const listener = alb.addListener('ListenerALB', { protocol: lbv2.ApplicationProtocol.HTTP, port: 8080 });
      const targetGroup = listener.addTargets('Fleet', { protocol: lbv2.ApplicationProtocol.HTTP, port: 8080 });
      loadBalancers.push(codedeploy.LoadBalancer.application(targetGroup));

      loadBalancerInfo.TargetGroupInfoList?.push({
        'Name': {
          'Fn::GetAtt': [
            'ALBListenerALBFleetGroupF4CBAA91',
            'TargetGroupName',
          ],
        },
      });
    }

    if (withNLB) {
      const nlb = new lbv2.NetworkLoadBalancer(stack, 'NLB', {
        vpc: defaultVpc,
      });

      const nlbListener = nlb.addListener('ListenerNLB', { port: 8080, protocol: lbv2.Protocol.TCP });
      const nlbTargetGroup = nlbListener.addTargets('Fleet', { port: 8080, protocol: lbv2.Protocol.TCP });
      loadBalancers.push(codedeploy.LoadBalancer.network(nlbTargetGroup));

      loadBalancerInfo.TargetGroupInfoList?.push({
        'Name': {
          'Fn::GetAtt': [
            'NLBListenerNLB0FleetGroup657E42E4',
            'TargetGroupName',
          ],
        },
      });
    }

    new codedeploy.ServerDeploymentGroup(stack, 'DeploymentGroup', {
      loadBalancers,
    });

    Template.fromStack(stack).hasResourceProperties('AWS::CodeDeploy::DeploymentGroup', {
      'LoadBalancerInfo': loadBalancerInfo,
      'DeploymentStyle': {
        'DeploymentOption': 'WITH_TRAFFIC_CONTROL',
      },
    });
  });
```
Thanks for the effort of considering all possible cases and producing a comprehensive test case. 💪
This is my suggestion to keep this more concise and maintainable while testing for all possible combinations of multiple LBs.
Feel free to adjust it or suggest improvements.

Reply:
Yes, this is a lot more concise! Thank you for taking a look!

Should we add the case where there could me more than one of the same type of LB?

Reply:
I think that with this implementation we're already covering all scenarios.
Adding LBs of the same type vs adding LBs of different types should not make a difference for the scope of this test.

Reply:
That's a faire point, all the paths are still covered in your suggested change, it's also cleaner. Added the modification